src/utils/url-validator.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Secure URL Validation Utilities
+ *
+ * Provides proper URL parsing and domain validation to prevent:
+ * - Domain hijacking (a-grafana.com matching grafana.com)
+ * - Path injection (evil.com/grafana.com/docs/)
+ * - Subdomain hijacking (grafana.com.evil.com)
+ * - Protocol bypasses (file://, data:, javascript:)
+ */
+
+/**
+ * Safely parse a URL string, returning null on failure
+ *
+ * @param urlString - The URL string to parse
+ * @returns Parsed URL object or null if invalid
+ */
+export function parseUrlSafely(urlString: string): URL | null {
+  try {
+    return new URL(urlString);
+  } catch {
+    return null;
+  }
+}
+
+/**
+ * Check if URL is a Grafana documentation URL
+ *
+ * Security: Validates hostname is exactly grafana.com or a proper subdomain
+ *
+ * @param urlString - The URL to validate
+ * @returns true if valid Grafana docs URL, false otherwise
+ *
+ * @example
+ * isGrafanaDocsUrl('https://grafana.com/docs/grafana/') // true
+ * isGrafanaDocsUrl('https://a-grafana.com/docs/') // false (domain hijacking)
+ * isGrafanaDocsUrl('https://grafana.com.evil.com/docs/') // false (subdomain hijacking)
+ */
+export function isGrafanaDocsUrl(urlString: string): boolean {
+  const url = parseUrlSafely(urlString);
+  if (!url) {
+    return false;
+  }
+
+  // Only allow http and https protocols
+  if (url.protocol !== 'http:' && url.protocol !== 'https:') {
+    return false;
+  }
+
+  // Check hostname is exactly grafana.com or a proper subdomain
+  // ✅ grafana.com
+  // ✅ www.grafana.com
+  // ✅ docs.grafana.com
+  // ❌ a-grafana.com (not grafana.com)
+  // ❌ grafana.com.evil.com (not a subdomain of grafana.com)
+  const isGrafanaDomain = url.hostname === 'grafana.com' || url.hostname.endsWith('.grafana.com');
+  if (!isGrafanaDomain) {
+    return false;
+  }
+
+  // Check pathname contains allowed documentation paths
+  // Learning journeys are at /docs/learning-journeys/ so we need includes(), not startsWith()
+  return (
+    url.pathname.startsWith('/docs/') ||
+    url.pathname.startsWith('/tutorials/') ||
+    url.pathname.includes('/learning-journeys/')
+  );
+}
+
+/**
+ * Check if URL is a valid YouTube domain
+ *
+ * Security: Validates hostname is an exact match to known YouTube domains
+ *
+ * @param urlString - The URL to validate
+ * @returns true if valid YouTube URL, false otherwise
+ *
+ * @example
+ * isYouTubeDomain('https://www.youtube.com/embed/abc') // true
+ * isYouTubeDomain('https://youtube.com.evil.com/embed/') // false
+ */
+export function isYouTubeDomain(urlString: string): boolean {
+  const url = parseUrlSafely(urlString);
+  if (!url) {
+    return false;
+  }
+
+  // Only allow http and https protocols
+  if (url.protocol !== 'http:' && url.protocol !== 'https:') {
+    return false;
+  }
```

Comment:
maybe only allow HTTPS for youtube?
```suggestion
  // Only allow https protocol
  if (url.protocol !== 'https:') {
    return false;
  }
```